packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -63,41 +62,32 @@ export class MonacoEditorProvider {
         @inject(ApplicationServer) protected readonly applicationServer: ApplicationServer,
         @inject(monaco.contextKeyService.ContextKeyService) protected readonly contextKeyService: monaco.contextKeyService.ContextKeyService
     ) {
-        const init = monaco.services.StaticServices.init.bind(monaco.services.StaticServices);
+        const staticServices = monaco.services.StaticServices;
+        const init = staticServices.init.bind(monaco.services.StaticServices);
         this.applicationServer.getBackendOS().then(os => {
             this.isWindowsBackend = os === OS.Type.Windows;
         });
+        if ('resourcePropertiesService' in staticServices) {
+            // tslint:disable-next-line:no-any
+            (staticServices['resourcePropertiesService'] as any)._factory = () => ({
+                getEOL: () => {
```

Comment:
I wonder should not we rather override [getOS](https://github.com/microsoft/vscode/blob/9f2d08eaa3c588fbdf49b07e394e357addef317b/src/vs/workbench/services/textfile/common/textResourcePropertiesService.ts#L41) and let Monaco do the rest. 

Our implementation looks bogus: it does not respect language and resource. And we already hook our preferences into Monaco configuration service.

Reply:
done